### PR TITLE
fix(delete account): update currentAccountId only if needed

### PR DESCRIPTION
**Pull request by bumi**
#### Type of change (Remove other not matching type)
- Bug fix (non-breaking change which fixes an issue)

#### Describe the changes you have made in this PR -
In the delete action we now only update the currentAccountId state if the current account was deleted.

#### Checklist:
- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

Would be nice if somebody could help me write a spec for this

### CODE REVIEW

**Comment by escapedcat** [comment]
> Would be nice if somebody could help me write a spec for this  

I was thinking if it might be better to test the background-code with intergration/e2e-tests instead of real unit-tests. With that we would test the whole interactivity for i.e. "delete account". The underlying code can change without the need of updating uni-tests.
So as soon as #627 is ready we can cover this code here as well.
What do you think @dylancom @bumi ?

**Comment by bumi** [comment]
I like both. often unit tests are easier to write, faster and the units are reused in many parts of the app.

**Comment by escapedcat** [comment]
Test via #747

**Comment by bumi** [comment]
thanks for the help with the tests @escapedcat

File: jest.config.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ module.exports = {
   transform: {
     "^.+\\.[t|j]sx?$": "babel-jest",
   },
+  transformIgnorePatterns: ["node_modules/(?!(@runcitadel))/"],
```
Comment (bumi) [inline jest.config.ts:6]: do we still need this?

**Review by bumi — COMMENTED** [review]
mergable? @escapedcat

File: jest.config.ts (hunk 1 above)
Comment (escapedcat) [reply jest.config.ts:6]: I believe so, yes

Comment (bumi) [reply jest.config.ts:6]: wasn't there a change in that package? 
(and I don't like such special cases :) )

Comment (escapedcat) [reply jest.config.ts:6]: Not sure, will check

Comment (escapedcat) [reply jest.config.ts:9]: My understanding is that runcitadel's dist code is coming as ESM (using import/export) which can't be handled by Jest. Therefor we need to transform the citadel code eventough normally we do not compile any code from `node_modules`.  
@dylancom what do you think about this?

Comment (bumi) [reply jest.config.ts:9]: ah I see. thanks for the explanation. 

but then we can merge this PR?

Comment (escapedcat) [reply jest.config.ts:9]: I think so, yes

**Pull request merged**